### PR TITLE
Finding no of set bits using existing inbuild library

### PR DESCRIPTION
**Pull request by deoliaman00**
To find the number of set bits in an efficient way. It consumes less time complexity than the previously given solution O(logn) solution.

### CODE REVIEW

File: bit_manipulation/count_of_set_bits2.cpp [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ * @details
+ * How to calculate the no of Set bits in a number.
+ * math equations, etc.
+ * @author [Aman Deoli](https://github.com/deoliaman00)
+ * @see related_file.cpp, another_file.cpp
+ */
+
+#include <cassert>   /// for assert
+#include <stdio.h>   //stdio.h is a header file which has the necessary information to include the input/output related functions in our program.
+#include <iostream>
```
Comment (Panquesito7) [inline bit_manipulation/count_of_set_bits2.cpp:14]: Use `iostream` only. `stdio.h` is used only in C.
```suggestion
#include <iostream>    /// for IO operations
```

File: bit_manipulation/count_of_set_bits2.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                        // no Overflow can be there.
+
+    return __builtin_popcount(n);
+    // Why this algorithm is better than the standard one?
+    // Because this algorithm takes O(k) time where k is no of set bits
+}
+}  // namespace no_of_set_bits
+}  // namespace bit_manipulation
+
+static void test() {
```
Comment (Panquesito7) [inline bit_manipulation/count_of_set_bits2.cpp:43]:
```suggestion
/**
 * @brief Self-test implementations
 * @returns void
 */
static void test() {
```

File: bit_manipulation/count_of_set_bits2.cpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    // n = 15 return 4
+    assert(bit_manipulation::no_of_set_bits::countSetBits(15) == 4);
+    // n = 25 return 3
+    assert(bit_manipulation::no_of_set_bits::countSetBits(25) == 3);
+    // n = 31 return 5
+    assert(bit_manipulation::no_of_set_bits::countSetBits(31) == 5);
+    // n = 32 return 1
+    assert(bit_manipulation::no_of_set_bits::countSetBits(32) == 1);
+    std::cout << "Test cases passed successfully" << std::endl;
+}
```
Comment (Panquesito7) [inline bit_manipulation/count_of_set_bits2.cpp:61]:
```suggestion
}

```

File: bit_manipulation/count_of_set_bits2.cpp [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+/**
+ * @brief Main function
+ * @returns 0 on exit
+ */
+int main() {
+    test();  // run self-test implementations
+    return 0;
+}
```
Comment (Panquesito7) [inline bit_manipulation/count_of_set_bits2.cpp:69]: Missing an endline. Without this endline, there could be issues with some IDEs and configurations.
```suggestion
}

```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Nice work! 🚀
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/deoliaman00/C-Plus-Plus/actions

File: bit_manipulation/count_of_set_bits2.cpp (hunk 4 above)
Comment (deoliaman00) [reply bit_manipulation/count_of_set_bits2.cpp:69]: Did you meant this 
int main() {
    test();  // run self-test implementations
    std::cout << std::endl;
    return 0; 
}

**Pull request closed without merging**